RetakesAllocator/Menus/GunsMenu.cs

```diff
@@ -30,7 +30,10 @@ private void OnMenuTimeout(CCSPlayerController player)
         player.PrintToChat($"{MessagePrefix}You did not interact with the menu in {MenuTimeout} seconds!");
 
         PlayersInMenu.Remove(player);
-        _menuTimeoutTimers[player].Kill();
+        if (_menuTimeoutTimers.TryGetValue(player, out var playerTimer))
+        {
+            playerTimer.Kill();
+        }
         _menuTimeoutTimers.Remove(player);
```

Comment:
Might be worth doing this in the if statement above

Reply:
went a slightly different direction